test/specs/test_wishlist_page.js

```diff
@@ -0,0 +1,123 @@
+const products = require('../data/products')
+const money_formatter = require('../formatters/money.formatter')
+const assert = require('assert')
+
+var product_quantity = [0, 0]
+describe('Test the wishlist', () => {
+    before(() => {
+        Wishlist.open()
+        Wishlist.setFrame()
+        Wishlist.enterFullScreen()
+    })
+
+    describe('Check status of the wishlist', () => {
+        it('15. should be empty when wishlist has no item', () => {
+            Wishlist.wishlist_tab.click()
+            expect(Wishlist.number_of_item_in_wishlist).toHaveText('(0)')
+            expect(Wishlist.message_when_empty_wishlist).toHaveText("The wishlist is empty!")
+        })
+    })
+    describe('Add/Remove/Update', () => {
+        beforeEach( () => {
+            product_quantity.pop()
+            product_quantity = [0, 0];
+            Wishlist.add_item_to_wishlist(products.products[0].name)
+            product_quantity[0]++
+            Wishlist.add_item_to_wishlist(products.products[1].name)
+            product_quantity[1] = product_quantity[1] + 2
+            Wishlist.open_wishlist_page()
+        })
+    
+        describe('Add same products to wishlist', () => {
+            it('16. should products are showed correctly in wishlist', () => {
+                Wishlist.add_item_to_wishlist(products.products[1].name)
+                product_quantity[1] = product_quantity[1] + 2
+                expect(Wishlist.wishlist_notification_message).toHaveText("The product has been added to your wishlist")
+                Wishlist.wishlist_notification_close_icon.click()
+                Wishlist.open_wishlist_page()
+                expect(Wishlist.number_of_item_in_wishlist).toHaveText('(' + (product_quantity[1] + product_quantity[0]) + ')')
+                expect(Wishlist.get_sku_value(1)).toHaveText(products.products[1].sku)
+                expect(Wishlist.get_product_name(1)).toHaveText(products.products[1].name)
+                expect(Wishlist.get_price(1)).toHaveText(money_formatter.usd_format.format(products.products[1].price))
+                expect(Wishlist.get_total_value(1)).toHaveText(money_formatter.usd_format.format(products.products[1].price*product_quantity[1]))
+                expect(Wishlist.get_quantity_of_item(1)).toHaveAttribute('value', product_quantity[1])
+            })
+        })
+    
+        describe('Add different products to wishlist', () => {
+            it('17. should new products are showed correctly in wishlist', () => {
+                Wishlist.add_item_to_wishlist(products.products[2].name)
+                product_quantity.push(1)
+                expect(Wishlist.wishlist_notification_message).toHaveText("The product has been added to your wishlist")
+                Wishlist.wishlist_notification_close_icon.click()
+                Wishlist.open_wishlist_page()
+                let total_of_product = 0
+                for (var i = 0; i < product_quantity.length; i++) {
```

Comment:
Please try to use Array.reduce() instead of for loop.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Reply:
I fixed it